The is a note page for myself to record everything about SNP calling.

# General Workflow
Some link from the internet:
```
http://userweb.eng.gla.ac.uk/cosmika.goswami/snp_calling/SNPCalling.html
https://bcbio.wordpress.com/2013/10/21/updated-comparison-of-variant-detection-methods-ensemble-freebayes-and-minimal-bam-preparation-pipelines/
```

### Samtools
Manual:
```
http://www.htslib.org/doc/samtools.html
https://samtools.github.io/hts-specs/SAMv1.pdf
https://genome.sph.umich.edu/wiki/SAM
```

### VCFtools
Manual
```
http://vcftools.sourceforge.net/perl_module.html#vcf-consensus
https://samtools.github.io/hts-specs/VCFv4.2.pdf
http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/
```

### BCFtools
```
https://samtools.github.io/bcftools/bcftools.html
```

### BWA aligner
```
http://bio-bwa.sourceforge.net/bwa.shtml#4
```

### Cufflink
```
http://www.cureffi.org/2013/11/18/an-mrna-seq-pipeline-using-gsnap-samtools-cufflinks-and-bedtools/
```

### Stampy
```
http://www.well.ox.ac.uk/~gerton/README.txt
```

### Mummer
```
http://mummer.sourceforge.net/
```

### Lastz
```
http://www.bx.psu.edu/~rsharris/lastz/README.lastz-1.04.00.html
```
